File: U1/FormularioV2/FormularioV2/menu.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace FormularioV2
{
    public partial class menu : Form
    {
        Int16 opcion;
        decimal[] recuperado = new decimal[3];
        decimal[] respuesta = new decimal[5];
        decimal sum = 0, resta = 0, multi = 0, div = 0;
        bool band1 = false, band2 = false, band3 = false, band4 = false;
        public menu()
        {
            InitializeComponent();
        }

        private void txtOpcion_TextChanged(object sender, EventArgs e)
        {
            if(txtOpcion.Text != "")
            {
                opcion = Convert.ToInt16(txtOpcion.Text);
            }
        }

       
        private void btnAceptar_Click(object sender, EventArgs e)
        {
            Menu.menu objMenu = new Menu.menu();
            switch (opcion)
            {
                case 1:
                    this.Hide();
                    objMenu.Show();
                    break;
                case 2:
                    MessageBox.Show("Suma realizada");
                    band1 = true;
                    respuesta[0] = suma(recuperado, band1);
                    break;
                case 3:
                    MessageBox.Show("Resta realzada");
                    band2 = true;
                    respuesta[1] = restita(recuperado, band2);
                    
                    break;
                case 4:
                    MessageBox.Show("Multiplicacion realizada");
                    band3 = true;
                    respuesta[2] = multiplicacion(recuperado, band3);
                    
                    break;
                case 5:
                    MessageBox.Show("Division realizada");
                    band4 = true;
                    break;
                case 6:
                    break;
                default:
                    txtOpcion.Text = "";
                    //Aqui se imprime las respuesta guardadas en el arreglo
                    MessageBox.Show("Querido Usuario lea las opciones");
                    break;
            }
        }


        public decimal[] recuperarInfo(decimal[] algunValor)
        {
            for(Int16 i = 0; i < algunValor.Length; i++)
            {
                recuperado[i] = algunValor[1];
            }

            return recuperado;
        }

        public decimal suma(decimal[] arraySuma, bool band)
        {
            if(band == true)
            {
                for (Int16 i = 0; i < arraySuma.Length; i++)
                {
                    sum += arraySuma[i];
                }
            }
            else
            {
                sum = 0;
            }
            
            return sum;
        }

        public decimal restita(decimal[] arrayResta, bool band)
        {
            if(band == true)
            {
                for (Int16 i = 0; i < arrayResta.Length; i++)
                {
                    resta -= arrayResta[i];
                }
            }
            else
            {
                resta = 0;
            }
            
            return resta;
        }

        public decimal multiplicacion(decimal[] arrayMulti, bool band)
        {
            if(band == true)
            {
                for (Int16 i = 0; i < arrayMulti.Length; i++)
                {
                    multi *= arrayMulti[i];
                }
            }
            else
            {
                multi = 0;
            }
            
            return multi;
        }
    }
}

File: U2/Tarea-02/Tarea-02/Tarea-02/Menu.vb
Public Class Menu
    'Opcion del menu
    Dim opcion As Int16

    'Recuperar los valores
    Dim val1 As Int16
    Dim val2 As Int16
    Dim val3 As Int16

    Dim sum As Int16 = 0
    Dim rest As Int16 = 0
    Dim mult As Int16 = 0

    Dim ban1 As Boolean = False
    Dim ban2 As Boolean = False
    Dim ban3 As Boolean = False

    Private Sub txtOption_TextChanged(sender As Object, e As EventArgs) Handles txtOption.TextChanged
        If txtOption.Text <> "" Then
            opcion = Convert.ToInt16(txtOption.Text)

        End If

    End Sub

    Private Sub btnAceptar_Click(sender As Object, e As EventArgs) Handles btnAceptar.Click
        Dim objeto As Captura.Captura = New Captura.Captura()
        Dim objeto2 As Resultados.Form1 = New Resultados.Form1

        Select Case opcion
            Case 1
                Me.Hide()
                objeto.Show()
            Case 2
                MessageBox.Show("Suma realizada")
                ban1 = True
            Case 3
                MessageBox.Show("resta realizada")
                ban2 = True
            Case 4
                MessageBox.Show("multiplicacion realizada")
                ban3 = True
            Case 5
                MessageBox.Show("division realizada")
            Case 6
                Me.Hide()
                objeto2.Show()

            Case Else
                MessageBox.Show("Solo estan disponibles las opciones 1-6")


        End Select
    End Sub

    'Metodos para los calculos
    Public Sub recuperarInfo(ByVal a As Int16, ByVal b As Int16, ByVal c As Int16)
        val1 = a
        val2 = b
        val3 = c
    End Sub

    Public Function suma(ByVal a As Int16, ByVal b As Int16, ByVal c As Int16) As Int16
        sum = a + b + c

        Return sum
    End Function

    Public Function resta(ByVal a As Int16, ByVal b As Int16, ByVal c As Int16) As Int16
        rest = a + b + c

        Return rest
    End Function

    Public Function multi(ByVal a As Int16, ByVal b As Int16, ByVal c As Int16) As Int16
        mult = a * b * c
        Return mult
    End Function
End Class
